### ISSUE TITLE
[SPIKE][P2P] Peer Discovery and Dynamic Peer Addition/Removal - Research Gemini Algorithm

### ISSUE THREAD

**Issue by jessicadaugherty** [pokt-network/pocket#300]
## Objective

Research Gemini to determine if and/or how we apply the functionality of this algorithm in V1 P2P peer discovery and dynamic peer addition/removal as part of the Raintree algorithm implementation. Ideally done in parallel with LibP2P research. 

## Origin Document

[V1 P2P discovery](https://docs.google.com/document/d/10rHFr8HpUfpmHzXbyZWzq60aJDY1lYMR9kHNgdfgbNE/edit#heading=h.e1rj36wlfr45)

[Gemini: Probabilistic Routing Algorithm in Structured P2P Overlay] (https://www.researchgate.net/publication/221601988_Gemini_Probabilistic_Routing_Algorithm_in_Structured_P2P_Overlay)

## Goals

- Summarize research learnings and recommendations on leveraging the Gemini algorithm and next steps (if any)

## Deliverable

- Determine if we should use the Gemini algorithm for peer discovery and dynamic peer addition/removal as part of the Raintree implementation
- Determine if Gemini is compatible with LibP2P (if we choose to use that library)
- Research and document the Gemini algorithm as it pertains to Raintree and LibP2P in a 1-2 pager that includes:
   - What elements of the library should or should not be used
   - Implementation plans

## Non-goals / Non-deliverables

- Implementing Gemini

## General issue deliverables

- [ ] Update the appropriate CHANGELOG
- [ ] Update any relevant READMEs (local and/or global)
- [ ] Update any relevant global documentation & references
- [ ] If applicable, update the source code tree explanation
- [ ] If applicable, add or update a state, sequence or flowchart diagram using [mermaid](https://mermaid-js.github.io/mermaid/)

## [Optional] Testing Methodology

- \__REPLACE_ME_: Make sure to update the testing methodology appropriately\_
- **Task specific tests**: `make ...`
- **All tests**: `make test_all`
- **LocalNet**: verify a `LocalNet` is still functioning correctly by following the instructions at [docs/development/README.md](https://github.com/pokt-network/pocket/tree/main/docs/development)

---

**Creator**: @jessicadaugherty
